controllers/argocd/argocd_controller.go

```diff
@@ -59,6 +59,8 @@ var log = logr.Log.WithName("controller_argocd")
 //+kubebuilder:rbac:groups=argoproj.io,resources=applications;appprojects,verbs=*
 //+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=*,verbs=*
 //+kubebuilder:rbac:groups="",resources=pods;pods/log,verbs=get
+//+kubebuilder:rbac:groups=template.openshift.io,resources=templates;templateinstances;templateconfigs,verbs=*
+//+kubebuilder:rbac:groups="oauth.openshift.io",resources=oauthclients,verbs=get;list;watch;create;delete;patch;update
 
```

Comment:
We should not need to add these rbac. The controller sa is the one creates/deletes oauth client. Why they are added now during the upgrade of keycloak? Please remove them if we can.

Reply:
while testing I realised that the latest version of argocd-operator fails to create oauthclient and templateinstance. This is a blocker for keycloak setup in OpenShift.

I think we missed it somewhere during the operator-sdk migration or something else.

Reply:
We need automated tests to cover this.  A simple smoke test would've caught it.

Reply:
added e2e tests